pkg/config/config.go

```diff
@@ -221,19 +232,51 @@ func InitCliConfig(configAndStacksInfo schema.ConfigAndStacksInfo, processStacks
 		}
 	}
 
-	// https://gist.github.com/chazcheadle/45bf85b793dea2b71bd05ebaa3c28644
-	// https://sagikazarmark.hu/blog/decoding-custom-formats-with-viper/
+	// Unmarshal, process environment variables, imports, and command-line arguments as needed.
 	err = v.Unmarshal(&cliConfig)
 	if err != nil {
 		return cliConfig, err
 	}
 
-	// Process ENV vars
 	err = processEnvVars(&cliConfig)
 	if err != nil {
 		return cliConfig, err
 	}
 
+	// Check if no imports are defined
+	if len(cliConfig.Import) == 0 {
+		basePath, err := filepath.Abs(cliConfig.BasePath)
+		if err != nil {
+			return cliConfig, err
+		}
+		// Check for an `atmos.d` directory and load the configs if found
+		atmosDPath := filepath.Join(basePath, "atmos.d")
+		// Ensure the joined path doesn't escape the intended directory
+		if !strings.HasPrefix(atmosDPath, basePath) {
+			return cliConfig, fmt.Errorf("invalid atmos.d path: attempted directory traversal")
+		}
```

Comment:
Is this necessary, if we're doing a glob below anyways?

I think we can just emit a debug message if no configs found.